common/src/main/java/com/wynntils/core/discord/DiscordManager.java

```diff
@@ -0,0 +1,118 @@
+    private static final int TICKS_PER_UPDATE = 5;
+
+    private CreateParams params;
+    private DiscordGameSDKCore core;
+    private Activity activity;
+
+    private int ticksUntilUpdate = 0;
+
+    public DiscordManager() {
+        super(List.of());
+    }
+
+    public void load() {
+        DiscordGameSDKCore.loadLibrary();
+        if (!isReady()) {
+            createCore();
+        }
+    }
+
+    public boolean isReady() {
+        return core != null && core.isOpen() && activity != null;
+    }
+
+    public void unload() {
+        if (!isReady()) return;
+        try {
+            activity.close();
+            activity = null;
+            core.close();
+            core = null;
+            params.close();
+            params = null;
+        } catch (GameSDKException e) {
+            if (e.getResult() == Result.TRANSACTION_ABORTED) {
+                // This occurs when player closes game and JVM exits before we can close the core
+                return;
+            }
+            WynntilsMod.error("Could not unload Discord Game SDK", e);
+        }
+    }
+
+    private void createCore() {
+        params = new CreateParams();
+        try {
+            params.setClientID(DISCORD_APPLICATION_ID);
+            params.setFlags(CreateParams.getNoRequireDiscordFlags());
+            core = new DiscordGameSDKCore(params);
+            activity = new Activity();
+            activity.timestamps().setStart(Instant.now());
+        } catch (Throwable e) {
+            if (e instanceof GameSDKException gameSDKException
+                    && gameSDKException.getResult() == Result.INTERNAL_ERROR) {
+                // This occurs when player closes game and JVM exits before we can close the core
+                return;
+            }
+            WynntilsMod.error("Could not initialize Discord Game SDK", e);
+        }
+    }
+
+    public void setDetails(String details) {
+        if (!isReady()) return;
+        activity.setDetails(details);
+        core.activityManager().updateActivity(activity);
+    }
+
+    public void setImage(String imageId) {
+        if (!isReady()) return;
+        activity.assets().setLargeImage(imageId);
+        core.activityManager().updateActivity(activity);
+    }
+
+    public void setImageText(String text) {
+        if (!isReady()) return;
+        activity.assets().setLargeText(text);
+        core.activityManager().updateActivity(activity);
+    }
+
+    public void setWynncraftLogo() {
+        if (!isReady()) return;
+        activity.assets().setLargeImage("wynn");
+        activity.assets().setLargeText("play.wynncraft.com");
+        core.activityManager().updateActivity(activity);
+    }
+
+    public void setState(String state) {
+        if (!isReady()) return;
+        activity.setState(state);
+        core.activityManager().updateActivity(activity);
+    }
+
+    @SubscribeEvent
+    public void onTick(TickEvent event) {
+        if (!isReady() || ticksUntilUpdate > 0) return;
```

Comment:
Maybe do ticksUntilUpdate-- somewhere here..? :-)

Also, please check ticksUntilUpdate first, since it is quickest. isReady() calls `.core.isOpen`. This sounds like it might be expensive; at the very least we should only check it once the TICKS_PER_UPDATE ticks have passed.